### PR TITLE
Add required=false option for nic devices with an unavailable parent

### PR DESCRIPTION
**Pull request by jeffreyxie0615**
Add required option to nic device interface that defaults to true. This option can be set using lxc network set <network_name> user.required={true/false}

### CODE REVIEW

File: lxd/device/nic_bridged.go [hunk 1]
```diff
@@ -479,8 +480,19 @@ func (d *nicBridged) PreStartCheck() error {
 		return nil
 	}
 
+	required := true
```
Comment (tomponline) [inline lxd/device/nic_bridged.go:483]: Let's use shared.IsTrueOrEmpty() here instead

**Review by tomponline — CHANGES_REQUESTED** [review]
From what I can sew it looks like if required=false and network is unavailable then device start will progress at normal which will still likely fail (because network is unavailable) so we will need to ensure device nic isnt actually started.

Also static analysis is failing.

**Comment by jeffreyxie0615** [comment]
If I'm understanding this correctly, PreStartCheck needs to return an error, but not actually cause the container to shutdown?

**Comment by tomponline** [comment]
> If I'm understanding this correctly, PreStartCheck needs to return an error, but not actually cause the container to shutdown?

The `disk` device currently has a similar `required` property. And that returns nil for PreStartCheck and Start when the source doesn't exist but required=false. 

See https://github.com/lxc/lxd/blob/master/lxd/device/disk.go#L470

**Comment by tomponline** [comment]
@jeffreyxie0615 hey, how are you getting on with this?

**Comment by tomponline** [comment]
@jeffreyxie0615 shall I close this for now?

**Pull request closed without merging**